packages/starknet-snap/src/utils/logger.ts

```diff
@@ -32,35 +32,31 @@ export const emptyLog: LoggingFn = (
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   {};
 
-class Logger implements ILogger {
-  readonly log: LoggingFn;
+export class Logger implements ILogger {
+  log: LoggingFn;
 
-  readonly warn: LoggingFn;
+  warn: LoggingFn;
 
-  readonly error: LoggingFn;
+  error: LoggingFn;
 
-  readonly debug: LoggingFn;
+  debug: LoggingFn;
 
-  readonly info: LoggingFn;
+  info: LoggingFn;
 
-  readonly trace: LoggingFn;
+  trace: LoggingFn;
 
-  readonly #logLevel: LogLevel = LogLevel.OFF;
+  #logLevel: LogLevel = LogLevel.OFF;
 
-  constructor() {
-    this.init(LogLevel.OFF.toString());
+  set logLevel(level: LogLevel) {
+    this.#logLevel = level;
```

Comment:
what happens if we set a logLevel that is not valid ? Maybe we could use a superstruct ?

Reply:
log level is not from client side, no need